modules/account-lib/src/coin/stx/utils.ts

```diff
@@ -0,0 +1,156 @@
+ */
+export function removeHexPrefix(hex: string): string {
+  return stripHexPrefix(hex);
+}
+
+/**
+ * @param publicKeyHash
+ * @param hashMode
+ * @param transactionVersion
+ */
+function getAddressFromPublicKeyHash(
+  publicKeyHash: Buffer,
+  hashMode: AddressHashMode,
+  transactionVersion: TransactionVersion,
+): string {
+  if (publicKeyHash.length !== 20) {
+    throw new Error('expected 20-byte pubkeyhash');
+  }
+
+  const addrVer = addressHashModeToVersion(hashMode, transactionVersion);
+  const addr = addressFromVersionHash(addrVer, publicKeyHash.toString('hex'));
+  const addrString = addressToString(addr);
+  return addrString;
+}
+
+/**
+ * @param tx
+ */
+export function getTxSenderAddress(tx: StacksTransaction): string {
+  if (tx.auth.spendingCondition !== null && tx.auth.spendingCondition !== undefined) {
+    const spendingCondition = tx.auth.spendingCondition;
+    const txSender = getAddressFromPublicKeyHash(
+      Buffer.from(spendingCondition.signer, 'hex'),
+      spendingCondition.hashMode as number,
+      tx.version,
+    );
+    return txSender;
+  } else throw new Error('spendingCondition should not be null');
+}
+
+/**
+ * Returns whether or not the string is a valid amount number
+ *
+ * @param {string} amount - the string to validate
+ * @returns {boolean} - the validation result
+ */
+export function isValidAmount(amount: string): boolean {
+  const bigNumberAmount = new BigNumber(amount);
+  return bigNumberAmount.isInteger() && bigNumberAmount.isGreaterThanOrEqualTo(0);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol address
+ *
+ * @param {string} address - the address to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidAddress(address: string): boolean {
+  return validateStacksAddress(address);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol transaction id or not.
+ *
+ * A valid transaction id is a SHA-512/256 hash of a serialized transaction; see
+ * the txidFromData function in @stacks/transaction:
+ * https://github.com/blockstack/stacks.js/blob/master/packages/transactions/src/utils.ts#L97
+ *
+ * @param {string} txId - the transaction id to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidTransactionId(txId: string): boolean {
+  if (txId.length !== 64 && txId.length !== 66) return false;
+  const noPrefix = removeHexPrefix(txId);
+  if (noPrefix.length !== 64) return false;
+
+  return allHexChars(noPrefix);
+}
+
+/**
+ * Returns whether or not the string is a valid protocol public key
+ *
+ * The key format is documented at
+ * https://github.com/stacksgov/sips/blob/main/sips/sip-005/sip-005-blocks-and-transactions.md#transaction-authorization
+ *
+ * @param {string} pub - the  public key to be validated
+ * @returns {boolean} - the validation result
+ */
+export function isValidPublicKey(pub: string): boolean {
+  if (pub.length !== 66 && pub.length !== 130) return false;
```

Comment:
There's no check for actual key validity here (only key format), so this will accept invalid (off curve) public keys. Not sure how much of an issue that is here, but it's worth calling out.